src/io/sb3/fromSb3.ts

```diff
@@ -1,67 +1,71 @@
 import * as JSZip from "jszip";
-import * as sb3 from "../sb3";
+import * as sb3 from "./interfaces";
 
 import { OpCode } from "../../OpCode";
 
 import Block, { BlockBase } from "../../Block";
 import * as BlockInput from "../../BlockInput";
 import Costume from "../../Costume";
-import List from "../../List";
 import Project from "../../Project";
 import Sound from "../../Sound";
 import { Sprite, Stage } from "../../Target";
-import Variable from "../../Variable";
+import { List, Variable } from "../../Data";
 import Script from "../../Script";
 
-type getAsset = (info: {
+type getAssetData = (info: {
   type: "costume" | "sound";
   name: string;
   md5: string;
   ext: string;
   spriteName: string;
 }) => Promise<any>;
 
-function extractCostumes(target: sb3.Target, getAsset: getAsset): Promise<Costume[]> {
+function extractCostumes(target: sb3.Target, getAssetData: getAssetData): Promise<Costume[]> {
   return Promise.all(
     target.costumes.map(
       async (costumeData: sb3.Costume) =>
         new Costume({
           name: costumeData.name,
-          md5: costumeData.assetId,
-          ext: costumeData.dataFormat,
-          bitmapResolution: 2,
-          centerX: costumeData.rotationCenterX * 2,
-          centerY: costumeData.rotationCenterY * 2,
-          asset: await getAsset({
+          dataFormat: costumeData.dataFormat,
+          data: await getAssetData({
             type: "costume",
             name: costumeData.name,
             md5: costumeData.assetId,
             ext: costumeData.dataFormat,
             spriteName: target.name
-          })
+          }),
+
+          md5: costumeData.assetId,
+          ext: costumeData.dataFormat,
+
+          bitmapResolution: costumeData.bitmapResolution || 1,
```

Comment:
Scratch 3.0 treats an unspecified `bitmapResolution` as 2.

Reply:
[Here's the code for reference](https://github.com/LLK/scratch-vm/blob/01cb244e3dc5296183ad67a6ee7cd80c4ca1b355/src/import/load-costume.js#L133). `scale` refers to the scaling factor, so costumes will only be scaled up to double resolution if they are explicitly stated to have a resolution of 1.